src/Widget/BaseForm.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Form\Widget;
+
+use InvalidArgumentException;
+use Yiisoft\Html\Html;
+use Yiisoft\Form\FormInterface;
+
+use function preg_match;
+
+final class BaseForm
```

Comment:
It is not a widget. I'd rename it to `FormHelper` or `FormHtml`.

Reply:
Done.